src/importer.rs

```diff
@@ -334,6 +241,278 @@ impl PyImporter {
+                        "SKIP" => invisible_solids = InvisibleSolids::Skip,
+                        _ => return Err(PyTypeError::new_err("unexpected kwarg value")),
+                    },
+                    "import_props" => {
+                        import_props = value.extract()?;
+                    }
+                    "import_entities" => {
+                        import_other_entities = value.extract()?;
+                    }
+                    "import_sky" => {
+                        import_skybox = value.extract()?;
+                    }
+                    "scale" => {
+                        scale = value.extract()?;
+                    }
+                    // Known importer-wide settings, MDL-specific settings, and special filesystem settings - ignore here
+                    "import_materials"
+                    | "simple_materials"
+                    | "allow_culling"
+                    | "editor_materials"
+                    | "texture_format"
+                    | "texture_interpolation"
+                    | "import_lights"
+                    | "light_factor"
+                    | "sun_factor"
+                    | "ambient_factor"
+                    | "import_sky_camera"
+                    | "sky_equi_height"
+                    | "target_fps"
+                    | "remove_animations"
+                    | "import_unknown_entities"
+                    | "import_animations"
+                    | "vmf_path" | "map_data_path" | "root_search" => {}
+                    _ => unrecognized_keys.push(key_str),
+                }
+            }
+        }
+
+        if !unrecognized_keys.is_empty() {
+            return Err(PyTypeError::new_err(format!(
+                "unexpected VMF kwargs: {}",
+                unrecognized_keys.join(", ")
+            )));
+        }
+
+        Ok(VmfSettings {
+            import_brushes,
+            import_overlays,
+            epsilon,
+            cut_threshold,
+            merge_solids,
+            invisible_solids,
+            import_props,
+            import_other_entities,
+            import_skybox,
+            scale,
+        })
+    }
+
+    /// Extract MDL-specific settings with old parameter names
+    pub fn extract_mdl_settings(kwargs: Option<&PyDict>) -> PyResult<bool> {
+        let mut import_animations = true;
+        let mut unrecognized_keys = Vec::new();
+
+        if let Some(kwargs) = kwargs {
+            for (key, value) in kwargs {
+                if value.is_none() {
+                    continue;
+                }
+
+                let key_str: &str = key.extract()?;
+                match key_str {
+                    "import_animations" => import_animations = value.extract()?,
+                    // Known importer-wide settings, special filesystem settings, and VMF-specific settings - ignore here
+                    "import_materials"
```

Comment:
Could we have separate function for checking if the key is unknown, by having the _ case call that function? This way these don't need to be repeated everywhere.